<a href="https://colab.research.google.com/github/Lineker98/Pro-Ativa/blob/main/Results%20analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


## Análise K-Means

In [ ]:
import plotly.graph_objects as go
import plotly.express as px
from plotly.subplots import make_subplots
import plotly.offline as py
import pandas as pd
import numpy as np

In [ ]:
df_clustered = pd.read_csv('/content/drive/MyDrive/UFOP/Pro Ativa/Dados/df_clustered.csv')

In [ ]:
df_clustered.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 132 entries, 0 to 131
Data columns (total 47 columns):
 #   Column                                                                 Non-Null Count  Dtype  
---  ------                                                                 --------------  -----  
 0   Sexo                                                                   132 non-null    int64  
 1   IdadeEvasao                                                            132 non-null    float64
 2   PeriodoEvasao                                                          132 non-null    int64  
 3   baixorendimentonasdisciplinas                                          132 non-null    float64
 4   cenáriopolíticoeconômicoinstável                                       132 non-null    float64
 5   desinteressecomcursoescolhido                                          132 non-null    float64
 6   desinteressecomcursosuperioremgeral                                    132 non-null    flo

In [ ]:
df_componentes = df_clustered.iloc[:, 32:]

In [ ]:
df_componentes.head()

PC1       PC2       PC3  ...      PC13      PC14  Cluster
0 -1.060073  0.385629 -0.184174  ... -0.021259 -0.024923        0
1 -0.470921  1.597739 -0.178995  ... -0.217172  0.262791        0
2  0.579732  0.928002 -0.204167  ...  0.354879  0.162148        1
3  0.930779 -0.177392  1.208637  ...  0.495717  0.185151        1
4 -0.307282 -0.562712 -0.153217  ...  0.045125  0.028173        0

[5 rows x 15 columns]

In [ ]:
componentes = ['PC' + str(i)  for i in range(1, 15)]
print(componentes)

['PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'PC6', 'PC7', 'PC8', 'PC9', 'PC10', 'PC11', 'PC12', 'PC13', 'PC14']


In [ ]:
fig = make_subplots(rows=2, cols=1, subplot_titles=('Distribuição das Componentes no Cluster 0', 'Distribuição das componentes no cluster 1'))

for componente in componentes:
    fig.append_trace(go.Violin(
                            y=df_componentes[componente][ df_componentes['Cluster'] == 0],
                            name=componente,
                            meanline_visible=True,
                            legendgroup=componente, scalegroup=componente,
                            ), row=1, col=1)
    fig.append_trace(go.Violin(
                            y=df_componentes[componente][ df_componentes['Cluster'] == 1],
                            name=componente,
                            meanline_visible=True,
                            legendgroup=componente, scalegroup=componente,

                            ), row=2, col=1)
    
fig.update_layout(height=800, width=1200,
                  showlegend=False, violingap=0,
                  violinmode='overlay')
fig.show()

In [ ]:
fig = go.Figure()

for componente in componentes:
    fig.add_trace(go.Violin(
                            y=df_componentes[componente][ df_componentes['Cluster'] == 0],
                            name=componente,
                            meanline_visible=True,
                            legendgroup=componente, scalegroup=componente,
                            side='negative'
                            ))
    fig.add_trace(go.Violin(
                            y=df_componentes[componente][ df_componentes['Cluster'] == 1],
                            name=componente,
                            meanline_visible=True,
                            legendgroup=componente, scalegroup=componente,
                            side='positive'
                            ))
    
fig.update_layout(height=800, width=1200,
                  showlegend=False, violingap=0,
                  violinmode='overlay', title='Distribuição das respostas dos usuários obtidas pelo PCA em cada componente em cada cluster')
fig.show()

In [ ]:
qtde_cluster = []
qtde_cluster.append(df_clustered['Cluster'][df_clustered['Cluster'] == 0].count())
qtde_cluster.append(df_clustered['Cluster'][df_clustered['Cluster'] == 1].count())

In [ ]:
labels = ['Cluster 0', 'Cluster 1']

fig = go.Figure(data=[go.Pie(labels=labels, values=qtde_cluster)])

colors = ['#B3093F', '#451531']


fig.update_layout(
    title_text="Quantidade de observações em cada cluster",
    titlefont = {'family': 'Times New Roman',
                                'size': 26,
                                'color': 'black'},
                  
    legend_font = {'family': 'Times New Roman',
                   'size': 26},
    )

fig.update_traces(marker=dict(colors=colors, line=dict(color='#000000', width=2)),
                  hoverinfo='label+percent', textfont_size=20)

fig.show()

Observamos que foram agrupadas 93 observações para cluster $0$ e 39 observações no cluster $1$.


In [ ]:
fig = go.Figure()

box1 = go.Violin(
              x = df_clustered['Cluster'],
              y = df_clustered['IdadeEvasao'],
              marker_color = 'indianred',
              box_visible=True
              )
            

data = [box1]

layout = go.Layout(
              title = 'Distribuição das idades nos clusters',
              titlefont = {'family': 'Arial',
                                'size': 22,
                                'color': 'black'},
              xaxis = {
                       'title': 'Clusters',
                       'color': 'black',
                       'tickfont': {
                           'size': 16,
                                    },
                       'title_font': {
                           'color': 'black',
                           'family': 'Times New Roman',
                           'size': 24,
                                    }
                       },
              yaxis = {
                       'title': 'Idade',
                       'color': 'black',
                       'tickfont': {
                           'size': 16,
                                    },
                       'title_font': {
                           'color': 'black',
                           'family': 'Times New Roman',
                           'size': 24,
                                    }
                       },
      

)

fig = go.Figure(data=data, layout=layout)
py.iplot(fig)

- Cluster $0$:  Contém a maior parte das observações, possui uma maior dispersão nas idades, além de possuir uma maior concentração em torno do primeiro quartil (19 anos) e mediana (22 anos) e de conter os outliers desta feature.
- Cluster $1$: Com a menor parte das observações, possui também uma menor dispersão das idades, sendas essas altamente concentradas em torno do primeiro quartil (20 anos) e terceiro quartil (23.75 anos). Ambos os clusters possuem a mesma mediana.

In [ ]:
fig = go.Figure()

violin = go.Violin(
              x = df_clustered['Cluster'],
              y = df_clustered['PeriodoEvasao'],
              marker_color = 'indianred',
              box_visible=True,
              )
            

data = [violin]

layout = go.Layout(
              title = 'Distribuição dos períodos de evasão nos clusters formados',
              titlefont = {'family': 'Times New Roman',
                                'size': 26,
                                'color': 'black'},
              xaxis = {
                       'title': 'Clusters',
                       'color': 'black',
                       'tickfont': {
                           'size': 18,
                                    },
                       'title_font': {
                           'color': 'black',
                           'family': 'Times New Roman',
                           'size': 26,
                                    }
                       },
              yaxis = {
                       'title': 'Períodos de Evasão',
                       'color': 'black',
                       'tickfont': {
                           'size': 18,
                                    },
                       'title_font': {
                           'color': 'black',
                           'family': 'Times New Roman',
                           'size': 26,
                                    }
                       },

)

fig = go.Figure(data=data, layout=layout)

py.iplot(fig)

- Análisando o gráfico acima, podemos notar que no cluster $0$, possuímos uma maior concentração de períodos de evasão em torno to primeiro quartil (1º período) e da mediana (segundo período), além de que, 75% das evasões ocorreram até o quarto período.

- Para o cluster $1$, que contém a menos observações, podemos observar uma maior dispersão nos períodos de evasão, com a maior concentração em torno do primeiro quartil (2º período) e terceiro quartil (5º período), além de possuir uma mediana de três.

### De acordo os atributos mais impotantes na primeira componente

In [ ]:
def organiza_dados(coluna: str) -> tuple:
  cluster_0 = df_clustered[coluna][df_clustered['Cluster'] == 0].value_counts().sort_index().tolist()
  cluster_1 = df_clustered[coluna][df_clustered['Cluster'] == 1].value_counts().sort_index().tolist()

  return (cluster_0, cluster_1)

In [ ]:
cluster_0, cluster_1 = organiza_dados('baixorendimentonasdisciplinas')

fig = make_subplots(rows=1, cols=2, specs=[[{'type':'domain'}, {'type':'domain'}]])

labels = ['Sem baixo rendimento', 'Com baixo rendimento']

fig.add_trace(go.Pie(labels=labels, values=cluster_0, name="Cluster 0"),
              1, 1)

fig.add_trace(go.Pie(labels=labels, values=cluster_1, name="Cluster 1"),
              1, 2)

fig.update_traces(hole=.4, hoverinfo="label+percent+name", textfont=dict(
        family="Times New Roman",
        size=22,
        color="black")
        )

fig.update_layout(
    
    title_text="Baixo rendimento nas disciplinas",
    titlefont = {'family': 'Times New Roman',
                                'size': 26,
                                'color': 'black'},

    legend = {
          'font': {
              'color': 'black',
              'family': 'Times New Roman',
              'size': 22,
          },
      },
  
    annotations=[dict(text='Cluster 0', x=0.18, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="Times New Roman",
                                                                                                size=16,
                                                                                                color="black")),
                 dict(text='Cluster 1', x=0.82, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="Times New Roman",
                                                                                                size=16,
                                                                                                color="black"))])

fig.show()

- O cluster $0$, que possui uma dispersão para os períodos de evasão mais concentrada em torno do primeiro e segundo período, e uma faixa etária de até 25 anos para $75$% dos alunos, teve como grande maioria alunos que não relataram baixo rendimento nas disciplinas.

- O cluster $1$, que possui os dados de períodos de evasão mais dispersos, tende a agrupar os alunos que alegaram terem tido baixo rendimento nas disciplinas.


In [ ]:
cluster_0, cluster_1 = organiza_dados('problemadeadaptaçãocomocurso')

fig = make_subplots(rows=1, cols=2, specs=[[{'type':'domain'}, {'type':'domain'}]])

labels = ['Sem problemas de adaptação', 'Com problemas de adaptação']

fig.add_trace(go.Pie(labels=labels, values=cluster_0, name="Cluster 0"),
              1, 1)

fig.add_trace(go.Pie(labels=labels, values=cluster_1, name="Cluster 1"),
              1, 2)

fig.update_traces(hole=.4, hoverinfo="label+percent+name", textfont=dict(
        family="sans serif",
        size=22,
        color="black")
        )

fig.update_layout(
    
    title_text="Problemas de adaptação em relação ao curso",

    titlefont = {
        'family': 'Times New Roman',
        'size': 28,
        'color': 'black'},

    legend = {
        'font': {
            'color': 'black',
            'family': 'Times New Roman',
            'size': 24,
        },
    },
                  
    annotations=[
                 dict(text='Cluster 0', x=0.17, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black")),
                 
                 dict(text='Cluster 1', x=0.83, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black"))])
fig.show()

- Cluster $0$: Maior parte das observações contidas não relataram problemas de adaptação com o curso.

- Cluster $1$: Cerca de $2/3$ das observações neste cluster relataram problemas de adaptação com o curso.

In [ ]:
cluster_0, cluster_1 = organiza_dados('dificuldadenaáreaescolhida')

fig = make_subplots(rows=1, cols=2, specs=[[{'type':'domain'}, {'type':'domain'}]])

labels = ['Sem dificuldades na área escolhida', 'Com dificuldades na área escolhida']

fig.add_trace(go.Pie(labels=labels, values=cluster_0, name="Cluster 0"),
              1, 1)

fig.add_trace(go.Pie(labels=labels, values=cluster_1, name="Cluster 1"),
              1, 2)

fig.update_traces(hole=.4, hoverinfo="label+percent+name", textfont=dict(
        family="sans serif",
        size=18,
        color="black")
        )

fig.update_layout(
    title_text="Dificuldade na área escolhida",

    titlefont = {'family': 'Times New Roman',
                                'size': 26,
                                'color': 'black'},
    legend = {
        'font': {
            'color': 'black',
            'family': 'Times New Roman',
            'size': 24,
        },
    },
                  
    annotations=[dict(text='Cluster 0', x=0.17, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black")),
                 dict(text='Cluster 1', x=0.83, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black"))])
fig.show()

- **Cluster $0$**: O mesmo cluster que tende a agrupar os alunos que não relatarm baixo rendimento nas disciplinas, cerca de $93$% destes mesmos alunos também não relataram dificuldade na área escolhida.

- **Cluster $1$**: Como podemos ver, cerca de $66.7$% dos alunos agrupados neste cluster relataram terem tido dificuldades na área escolhida, importante lembrar que, estes mesmo alunos, também tiveram em maioria baixo rendimento nas disciplians.

In [ ]:
cluster_0, cluster_1 = organiza_dados('problemascomaprendizadoemrelaçãoaoconteúdo')

fig = make_subplots(rows=1, cols=2, specs=[[{'type':'domain'}, {'type':'domain'}]])

labels = ['Sem problemas de aprendizagem', 'Com problemas de aprendizagem']

fig.add_trace(go.Pie(labels=labels, values=cluster_0, name="Cluster 0"),
              1, 1)

fig.add_trace(go.Pie(labels=labels, values=cluster_1, name="Cluster 1"),
              1, 2)

fig.update_traces(hole=.4, hoverinfo="label+percent+name", textfont=dict(
        family="sans serif",
        size=18,
        color="black"
    ))

fig.update_layout(
    
    title_text="Problemas de aprendizagem em relação ao conteúdo",
    titlefont = {'family': 'Times New Roman',
                                'size': 26,
                                'color': 'black'},

    legend = {
        'font': {
            'color': 'black',
            'family': 'Times New Roman',
            'size': 24,
        },
    },
    annotations=[dict(text='Cluster 0', x=0.17, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black")
                      ),
                 dict(text='Cluster 1', x=0.83, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black"))])
fig.show()

- **Cluster $0$**: Já podemos observar aqui que os alunos agrupados neste, tendem a não terem tido problemas em relação à aprendizagem no curso.

- **Cluster $1$**: Neste cluster, cerca $59$% dos alunos relataram terem tidos problemas de aprendizagem em relação ao conteúdo.

In [ ]:
cluster_0, cluster_1 = organiza_dados('problemaspsicológicos')

fig = make_subplots(rows=1, cols=2, specs=[[{'type':'domain'}, {'type':'domain'}]])

labels = ['Sem problemas psicológicos', 'Com problemas psicológicos']

fig.add_trace(go.Pie(labels=labels, values=cluster_0, name="Cluster 0"),
              1, 1)

fig.add_trace(go.Pie(labels=labels, values=cluster_1, name="Cluster 1"),
              1, 2)

fig.update_traces(hole=.4, hoverinfo="label+percent+name", textfont=dict(
        family="sans serif",
        size=18,
        color="black"
    ))

fig.update_layout(
    
    title_text="Problemas psicológicos",
    titlefont = {'family': 'Times New Roman',
                                'size': 26,
                                'color': 'black'},

    legend = {
        'font': {
            'color': 'black',
            'family': 'Times New Roman',
            'size': 24,
        },
    },

    annotations=[dict(text='Cluster 0', x=0.17, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black")
                      ),
                 dict(text='Cluster 1', x=0.83, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black"))])
fig.show()

### De acordo os atributos mais impotantes na segunda componente

In [ ]:
cluster_0, cluster_1 = organiza_dados('Sexo')

fig = make_subplots(rows=1, cols=2, specs=[[{'type':'domain'}, {'type':'domain'}]])

labels = ['Masculino', 'Feminino']

fig.add_trace(go.Pie(labels=labels, values=cluster_0, name="Cluster 0"),
              1,1)

fig.add_trace(go.Pie(labels=labels, values=cluster_1, name="Cluster 1"),
              1, 2)

fig.update_traces(hole=.4, hoverinfo="label+percent+name", textfont=dict(
        family="sans serif",
        size=18,
        color="black"
    ))

fig.update_layout(
    
    title_text="Problemas psicológicos",
    titlefont = {'family': 'Times New Roman',
                                'size': 26,
                                'color': 'black'},

    legend = {
        'font': {
            'color': 'black',
            'family': 'Times New Roman',
            'size': 24,
        },
    },

    annotations=[dict(text='Cluster 0', x=0.18, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black")
                      ),
                 dict(text='Cluster 1', x=0.82, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black"))])
fig.show()

In [ ]:
cluster_0, cluster_1 = organiza_dados('distanciamentodafamília')

fig = make_subplots(rows=1, cols=2, specs=[[{'type':'domain'}, {'type':'domain'}]])

labels = ['Não alegaram distanciamento', 'Alegaram distanciamento']

fig.add_trace(go.Pie(labels=labels, values=cluster_0, name="Cluster 0"),
              1, 1)

fig.add_trace(go.Pie(labels=labels, values=cluster_1, name="Cluster 1"),
              1, 2)

fig.update_traces(hole=.4, hoverinfo="label+percent+name", textfont=dict(
        family="sans serif",
        size=20,
        color="black"
    ))

fig.update_layout(
    title_text="Distanciamento da família",
    titlefont = {'family': 'Times New Roman',
                                'size': 26,
                                'color': 'black'},

    legend = {
            'font': {
            'color': 'black',
            'family': 'Times New Roman',
            'size': 24,
                    },
              },
    annotations=[dict(text='Cluster 0', x=0.18, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black")
                      ),
                 dict(text='Cluster 1', x=0.82, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black"))])
fig.show()

In [ ]:
cluster_0, cluster_1 = organiza_dados('problemasfinanceiros')

fig = make_subplots(rows=1, cols=2, specs=[[{'type':'domain'}, {'type':'domain'}]])

labels = ['Sem problemas financeiros', 'Com problemas financeiros']

fig.add_trace(go.Pie(labels=labels, values=cluster_0, name="Cluster 0"),
              1, 1)

fig.add_trace(go.Pie(labels=labels, values=cluster_1, name="Cluster 1"),
              1, 2)

fig.update_traces(hole=.4, hoverinfo="label+percent+name", textfont=dict(
        family="sans serif",
        size=20,
        color="black"
    ))

fig.update_layout(
    title_text="Problemas financeiros",
    titlefont = {'family': 'Times New Roman',
                                'size': 26,
                                'color': 'black'},
    legend = {
            'font': {
            'color': 'black',
            'family': 'Times New Roman',
            'size': 24,
                    },
              },
    annotations=[dict(text='Cluster 0', x=0.18, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black")
                      ),
                 dict(text='Cluster 1', x=0.82, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black"))])
fig.show()

In [ ]:
cluster_0, cluster_1 = organiza_dados('oportunidadedeingressoemoutrafaculdadecommesmocurso')

fig = make_subplots(rows=1, cols=2, specs=[[{'type':'domain'}, {'type':'domain'}]])

labels = ['Sem oportunidade', 'Com oportunidade']

fig.add_trace(go.Pie(labels=labels, values=cluster_0, name="Cluster 0"),
              1, 1)

fig.add_trace(go.Pie(labels=labels, values=cluster_1, name="Cluster 1"),
              1, 2)

fig.update_traces(hole=.4, hoverinfo="label+percent+name", textfont=dict(
        family="sans serif",
        size=18,
        color="black"
    ))

fig.update_layout(
    
    title_text="Oportunidade de ingresso em outra faculdade com o mesmo curso",
    titlefont = {'family': 'Times New Roman',
                                'size': 26,
                                'color': 'black'},
                  
    legend = {
            'font': {
            'color': 'black',
            'family': 'Times New Roman',
            'size': 24,
                    },
              },
    annotations=[dict(text='Cluster 0', x=0.18, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black")
                      ),
                 dict(text='Cluster 1', x=0.82, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black"))])
fig.show()

### Análises gerais

In [ ]:
cluster_0, cluster_1 = organiza_dados('problemascomprofessoresespecíficos')

fig = make_subplots(rows=1, cols=2, specs=[[{'type':'domain'}, {'type':'domain'}]])

labels = ['Sem problemas com professores', 'Com problemas com professores']

fig.add_trace(go.Pie(labels=labels, values=cluster_0, name="Cluster 0"),
              1, 1)

fig.add_trace(go.Pie(labels=labels, values=cluster_1, name="Cluster 1"),
              1, 2)

fig.update_traces(hole=.4, hoverinfo="label+percent+name", textfont=dict(
        family="sans serif",
        size=18,
        color="black"
    ))

fig.update_layout(
    title_text="Problemas com professores",
    titlefont = {'family': 'Times New Roman',
                                'size': 26,
                                'color': 'black'},

    legend = {
            'font': {
            'color': 'black',
            'family': 'Times New Roman',
            'size': 24,
                    },
              },

    annotations=[dict(text='Cluster 0', x=0.165, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black")
                      ),
                 dict(text='Cluster 1', x=0.835, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black"))])
fig.show()

In [ ]:
cluster_0, cluster_1 = organiza_dados('oportunidadedeingressoemoutrafaculdadecommesmocurso')

fig = make_subplots(rows=1, cols=2, specs=[[{'type':'domain'}, {'type':'domain'}]])                  

labels = ['Não relataram desinteresse', 'Relataram desinteresse']

fig.add_trace(go.Pie(labels=labels, values=cluster_0, name="Cluster 0"),
              1, 1)

fig.add_trace(go.Pie(labels=labels, values=cluster_1, name="Cluster 1"),
              1, 2)


fig.update_traces(hole=.4, hoverinfo="label+percent+name", textfont=dict(
        family="sans serif",
        size=18,
        color="black"
    ))

fig.update_layout(
    
    title_text="Desinteresse com o curso escolhido",
    titlefont = {'family': 'Times New Roman',
                                'size': 26,
                                'color': 'black'},

    legend = {
            'font': {
            'color': 'black',
            'family': 'Times New Roman',
            'size': 24,
                    },
              },

    annotations=[dict(text='Cluster 0', x=0.165, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black")
                      ),
                 dict(text='Cluster 1', x=0.835, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black"))])
fig.show()

In [ ]:
cluster_0, cluster_1 = organiza_dados('oportunidadedetrabalho')

fig = make_subplots(rows=1, cols=2, specs=[[{'type':'domain'}, {'type':'domain'}]])

labels = ['Sem oportunidade de trabalho', 'Com oportunidade de trabalho']

fig.add_trace(go.Pie(labels=labels, values=cluster_0, name="Cluster 0"),
              1, 1)

fig.add_trace(go.Pie(labels=labels, values=cluster_1, name="Cluster 1"),
              1, 2)


fig.update_traces(hole=.4, hoverinfo="label+percent+name", textfont=dict(
        family="sans serif",
        size=18,
        color="black"
    ))

fig.update_layout(
    title_text="Oportunidade de trabalho",
    titlefont = {'family': 'Times New Roman',
                                'size': 26,
                                'color': 'black'},

    legend = {
            'font': {
            'color': 'black',
            'family': 'Times New Roman',
            'size': 24,
                    },
              },

    annotations=[dict(text='Cluster 0', x=0.165, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black")
                      ),
                 dict(text='Cluster 1', x=0.835, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black"))])
fig.show()

In [ ]:
cluster_0, cluster_1 = organiza_dados('problemascomumadisciplinaemespecífico')

fig = make_subplots(rows=1, cols=2, specs=[[{'type':'domain'}, {'type':'domain'}]])
                    
labels = ['Não reletaram problemas', 'Relataram problemas']

fig.add_trace(go.Pie(labels=labels, values=cluster_0, name="Cluster 0"),
              1, 1)

fig.add_trace(go.Pie(labels=labels, values=cluster_1, name="Cluster 1"),
              1, 2)


fig.update_traces(hole=.4, hoverinfo="label+percent+name", textfont=dict(
        family="sans serif",
        size=18,
        color="black"
    ))

fig.update_layout(
    
    title_text="Problemas com uma disciplina em específico",
    titlefont = {'family': 'Times New Roman',
                                'size': 26,
                                'color': 'black'},

    legend = {
            'font': {
            'color': 'black',
            'family': 'Times New Roman',
            'size': 24,
                    },
              },

    annotations=[dict(text='Cluster 0', x=0.165, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black")
                      ),
                 dict(text='Cluster 1', x=0.835, y=0.5, font_size=22, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=20,
                                                                                                color="black"))])
fig.show()

In [ ]:
fig = make_subplots(rows=1, cols=2, specs=[[{'type':'domain'}, {'type':'domain'}]])

feminino = df_clustered['problemaspsicológicos'][df_clustered['Sexo'] == 0].value_counts().tolist()
masculino = df_clustered['problemaspsicológicos'][df_clustered['Sexo'] == 1].value_counts().tolist()

labels = ['Sem problemas psicológicos', 'Com problemas psicológicos']

fig.add_trace(go.Pie(labels=labels, values=feminino, name="Femino"),
              1, 1)

fig.add_trace(go.Pie(labels=labels, values=masculino, name="Masculino"),
              1, 2)


fig.update_traces(hole=.4, hoverinfo="label+percent+name", textfont=dict(
        family="sans serif",
        size=18,
        color="black"
    ))

fig.update_layout(
    title_text="Problemas psicológicos por para cada sexo",

    titlefont = {'family': 'Times New Roman',
                                'size': 26,
                                'color': 'black'},
    legend = {
            'font': {
            'color': 'black',
            'family': 'Times New Roman',
            'size': 24,
                    },
              },
                  
    annotations=[dict(text='Feminino', x=0.18, y=0.5, font_size=20, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=16,
                                                                                                color="black")
                      ),
                 dict(text='Masculino', x=0.82, y=0.5, font_size=20, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=16,
                                                                                                color="black"))])
fig.show()

In [ ]:
fig = go.Figure()

hist1 = go.Histogram(histfunc="avg", 
                     x=df_clustered['Cluster'],
                     y=df_clustered['positive_sent_pt'],
                     name="sentimento positivo pt",
                     marker_color='#6B8BFA')
hist2 = go.Histogram(histfunc="avg", 
                     x=df_clustered['Cluster'], 
                     y=df_clustered['positive_sent_en'], 
                     name="sentimento positivo ing", 
                     marker_color='#00004c')
hist3 = go.Histogram(histfunc="avg", 
                     x=df_clustered['Cluster'], 
                     y=df_clustered['negative_sent_pt'], 
                     name="sentimento negativo pt", 
                     marker_color='#FF4F4F')
hist4 = go.Histogram(histfunc="avg", 
                     x=df_clustered['Cluster'], 
                     y=df_clustered['negative_sent_en'], 
                     name="sentimento negativo ing", 
                     marker_color='#520004')

data = [hist1, hist2, hist3, hist4]

layout = go.Layout(
              title = 'Média dos sentimentos em cada cluster',
              titlefont = {'family': 'Times New Roman',
                                'size': 26,
                                'color': 'black'},

              legend = {
                  'font': {
                  'color': 'black',
                  'family': 'Times New Roman',
                  'size': 24,
                    },
              },
              xaxis = {
                       'title': 'Clusters',
                       'color': 'black',
                       'tickfont': {
                           'size': 20,
                                    },
                       'title_font': {
                           'color': 'black',
                           'family': 'Times New Roman',
                           'size': 26,
                                    }
                       },
              yaxis = {
                       'title': 'Intesidade de sentimento',
                       'color': 'black',
                       'tickfont': {
                           'size': 18,
                                    },
                       'title_font': {
                           'color': 'black',
                           'family': 'Times New Roman',
                           'size': 26,
                                    }
                       },


)

fig = go.Figure(data=data, layout=layout)
py.iplot(fig)

- Notemos que, ambos os clusters possuem um sentimento positivo para o inglês e o português próximo da neutralidade $(1)$. Contudo, o cluster $0$, que possui a maior quantidade observações, e dessas observações grande maioria não relataram problemas com o curso, este cluster possui uma média mais próxima da neutralidade de sentimento negativo para o inglês e português.
- Para o cluster $1$, que com a menor quantidade de observações, agrupa os alunos que relataram problemas no curso, notemos também que estes mesmo possuem um média mais distante da neutralidade, ou seja, um maior sentimento negativo relacionados á UFOP em geral.

In [ ]:
box1 = go.Box(
              x = df_clustered['Cluster'],
              y = df_clustered['positive_sent_en'],
              name = 'positive_sent_en',
              marker_color = 'green'
              )

box2 = go.Box(
              x = df_clustered['Cluster'],
              y = df_clustered['positive_sent_pt'],
              name = 'positive_sent_pt',
              marker_color = 'blue'
              )

box3 = go.Box(
              x = df_clustered['Cluster'],
              y = df_clustered['negative_sent_pt'],
              name = 'negative_sent_pt',
              marker_color = 'red'
              )

box4 = go.Box(
              x = df_clustered['Cluster'],
              y = df_clustered['negative_sent_en'],
              name = 'negative_sent_en',
              marker_color = 'purple'
              )

data = [box1, box2, box3, box4]

layout = go.Layout(
              title = 'Distribuição dos sentimentos em cada cluster',
              titlefont = {'family': 'Arial',
                                'size': 22,
                                'color': 'black'},
              xaxis = {'title': 'Clusters'},
              yaxis = {'title': 'Idade'},

)

fig = go.Figure(data=data, layout=layout)
fig.update_layout(
    boxmode='group'
)
py.iplot(fig)

In [ ]:
cluster_0, cluster_1 = organiza_dados('oportunidadedeingressoemoutrafaculdadeemcursodiferente')

fig = make_subplots(rows=1, cols=2, specs=[[{'type':'domain'}, {'type':'domain'}]])

labels = ['Sem oportunidade de ingresso', 'Com oportunidade de ingresso']

fig.add_trace(go.Pie(labels=labels, values=cluster_0, name="Cluster 0"),
              1, 1)

fig.add_trace(go.Pie(labels=labels, values=cluster_1, name="Cluster 1"),
              1, 2)


fig.update_traces(hole=.4, hoverinfo="label+percent+name", textfont=dict(
        family="sans serif",
        size=18,
        color="black"
    ))

fig.update_layout(
    
    title_text="Oportunidade de ingresso em outra faculdade em um curso diferente",
    titlefont = {'family': 'Times New Roman',
                                'size': 26,
                                'color': 'black'},
    legend = {
            'font': {
            'color': 'black',
            'family': 'Times New Roman',
            'size': 24,
                    },
              },
    annotations=[dict(text='Cluster 0', x=0.18, y=0.5, font_size=20, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=16,
                                                                                                color="black")
                      ),
                 dict(text='Cluster 1', x=0.82, y=0.5, font_size=20, showarrow=False, font=dict(
                                                                                                family="sans serif",
                                                                                                size=16,
                                                                                                color="black"))])
fig.show()

### Conclusão

Após a aplicação do PCA para redução de dimensionalidade dos nossos dados, e posteriormente a aplicação do algoritmo K-Means nas $132$ observações restantes, foi possível identificar a separação dos perfis de evasão dos alunos em duas classes. 

- 1ª classe (cluster 0):

  - Dentre as observações analisadas, podemos notar que cerca de $70.5$% foram agrupadas para o cluster $0$, o qual caracteriza a nossa primeira classe de perfis dos estudantes. As características que a compõem e que corresponde pela maioria dos estudantes analisados são:
    - Uma maior dispersão nas idade de evasão, sendo que $75$% das observações estão no limite de $25.25$ anos de idade.
    - Grande concentração de evasão entre o primeiro e segundo período, tendo também $75$% das observações até o quarto período.
    - Dos $93$ alunos desta classe, apenas uma pequena minoria relatou problemas com curso ou com o ICEA em geral como motivos de evasão, sendo o destaque negativo o atributo $\textit{desinteresse com o curso escolhido}$, com $24.7$% de alegações.
    - Um ponto válido de se observar nesta classe, é a grande quantidade de oportunidades de troca de cursos externas à UFOP ou (ICEA), ou de trabalho, cerca de ....
    - O que resultou em uma média de sentimento negativo mais próxima do neutro, e uma média de sentimento positivo mais alta.

- 2ª classe (cluster 1):

  - As observações restantes ($29.5$%) do total, foram agrupadas de forma que esta classe nos relata os perfis de alunos que desenvolveram problemas internos ao ICEA. Dos quais podemos destacar:

    - Nossa segunda classe possui uma menor dispersão das idades dos alunos, sendo altamente concentradas nos intervalos de $[21, 23.75]$ anos de idade.
    - Mesmo com a menor quantidade de observações, notamos uma maior variabilidade entre os período de evasão, tendo sua maior concentracão entre o primeiro quartil $2º$ período, e terceiro quartil $5º$ período.
    - Como descrito acima, este cluster descreve os alunos que tiveram problemas com seu ensino no ICEA, dentre eles, podemos citar: 
      - $84.6$% dos alunos neste cluster alegaram baixo rendimento nas disciplinas como motivo de evasão.
      - $66.7$% relataram dificuldades na área escolhida.
      - $59$% tiveram problemas na aprendizagem em relação ao conteúdo.
      - $48.7$% desenvolveram problemas psicológicos.
      - $35.9$% tiveram problemas com professores específicos.
      - $48.7$% relataram desinteresse com curso.
      - $43.6$ tiveram como motivo de evasão problemas em uma disciplina em específico.
    - O que resultou em uma média de sentimentos negativo mais distante do neutro, tanto para o inglês como português, por outro lado, uma média baixa de sentimento positivo.
